src/migration-scripts/ipsec/9-to-10

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2022 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from sys import argv
+from sys import exit
+
+from vyos.configtree import ConfigTree
+from vyos.template import is_ipv4
+from vyos.template import is_ipv6
+
+
+if (len(argv) < 1):
+    print("Must specify file name!")
+    exit(1)
+
+file_name = argv[1]
+
+with open(file_name, 'r') as f:
+    config_file = f.read()
+
+base = ['vpn', 'ipsec']
+config = ConfigTree(config_file)
+
+if not config.exists(base):
+    # Nothing to do
+    exit(0)
+
+# IKE changes, T4118:
+if config.exists(base + ['ike-group']):
+    for ike_group in config.list_nodes(base + ['ike-group']):
+        # replace 'ipsec ike-group XXX mobike disable'
+        #      => 'ipsec ike-group XXX disable-mobike'
+        mobike = base + ['ike-group', ike_group, 'mobike']
+        if config.exists(mobike):
+            if config.return_value(mobike) == 'disable':
+                config.set(base + ['ike-group', ike_group, 'disable-mobike'])
+            config.delete(mobike)
+
+        # replace 'ipsec ike-group XXX ikev2-reauth yes'
+        #      => 'ipsec ike-group XXX ikev2-reauth'
+        reauth = base + ['ike-group', ike_group, 'ikev2-reauth']
+        if config.exists(reauth):
+            if config.return_value(reauth) == 'yes':
+                config.delete(reauth)
+                config.set(reauth)
+            else:
+                config.delete(reauth)
+
+# ESP changes
+# replace 'ipsec esp-group XXX compression enable'
+#      => 'ipsec esp-group XXX enable-compression'
+if config.exists(base + ['esp-group']):
+    for esp_group in config.list_nodes(base + ['esp-group']):
+        compression = base + ['esp-group', esp_group, 'compression']
+        if config.exists(compression):
+            if config.return_value(compression) == 'enable':
+                config.set(base + ['esp-group', esp_group, 'enable-compression'])
+            config.delete(compression)
+
+# PEER changes
+if config.exists(base + ['site-to-site', 'peer']):
```

Comment:
This implementation seems good, but it's incomplete in one critical aspect: it doesn't handle `@id` peers.

If you do `set vpn ipsec site-to-site peer @somepeer`, it creates a peer that is allowed to connect from any address as long as it sends `somepeer` as its ID in IKE packets.

So if peer name looke like `@somepeer`, we need to migrate the part after `@` to `vpn ipsec site-to-site peer @somepeer authentication remote-id somepeer`.

The @-syntax for peers identified by IKE IDs is the main design flaw and a source of confusion for anyone who wants to allow a peer to connect from a non-fixed address. Migrating it is required to make the peer name purely informational and to remove that confusion.

Reply:
Done!